File: TestProject1/Endpoints/StudentEndPoints.cs
using RestSharp;
using TestProject1.Payload;

namespace TestProject1.Endpoints
{
    public class StudentEndPoints
    { 
        private readonly RestClient _client;
        private readonly string _baseurl;
        public StudentEndPoints(string baseurl)
        {
            _baseurl = baseurl;
            _client = new RestClient(_baseurl); 
        }
        
        public RestResponse CreateStudent(Student student)
        {
            var request = new RestRequest(Routes.common_student_endpoint, Method.Post);
            request.AddJsonBody(student);
            var response = _client.ExecutePost(request);
            return response;

        }

        public RestResponse ReadStudent()
        {
            var request = new RestRequest(Routes.common_student_endpoint, Method.Get);
            var response = _client.ExecuteGet(request);
            return response;
        }

        public RestResponse ReadStudentById(int id)
        {
            var request = new RestRequest($"{Routes.common_student_endpoint}/{id}", Method.Get);
            var response = _client.ExecuteGet(request);
            return response;
        }

        public RestResponse UpdateStudent(int id, Student student)
        {
            var request = new RestRequest($"{Routes.common_student_endpoint}/{id}", Method.Put);
            request.AddJsonBody(student);
            var response = _client.ExecutePut(request);
            return response;
        }

        public RestResponse DeleteStudent(int id)
        {
            var request = new RestRequest($"{Routes.common_student_endpoint}/{id}", Method.Delete);
            var response = _client.ExecuteDelete(request);
            return response;
        }
    }
}

File: StudentApiAutomation/src/test/java/studentapi/endpoint/StudentEndPoints.java
package studentapi.endpoint;

import static io.restassured.RestAssured.given;

import io.restassured.http.ContentType;
import io.restassured.response.Response;
import studentapi.payload.Student;

public class StudentEndPoints {
	
	public static Response createStudent(Student payload) {
		Response response = 
				given()
					.contentType(ContentType.JSON)
					.accept(ContentType.JSON)
					.body(payload)
				.when()
					.post(Routes.post_student_url);
		return response;
		
	}
	
	public static Response readStudentById(int Id) {
		Response response =
				given()
				.pathParam("Id",Id)
				.when()
				.get(Routes.get_student_url);
		return response;
	}
	
	public static Response readAllStudent() {
		Response response = 
				given()
				.get(Routes.get_all_student_url);
		return response;
	}
	
	public static Response updateStudent(int Id, Student payload) {
		Response response = 
				given()
					.contentType(ContentType.JSON)
					.accept(ContentType.JSON)
					.pathParam("Id",Id)
					.body(payload)
				.when()
					.put(Routes.update_student_url);
		return response;
	}
	
	public static Response deleteStudent(int Id)
	{
		Response response = 
				given()
				 .pathParam("Id", Id)
		        .when()
		         .delete(Routes.delete_student_url);
		 return response;
	}

}
